### PR TITLE
Gcloud batch downloads

### PR DESCRIPTION
**Pull request by jburos**
Minor update to gcloud io utilities (in development) to allow user to customize certain options. 

1. set chunk_size to enable larger file downloads 
2. user can set tmpdir (when downloading to to a mounted fs, tmpdir should be on same fs)
3. pass kwargs to list_files -- useful for setting `prefix=`

### CODE REVIEW

File: cohorts/cohort.py [hunk 1]
```diff
@@ -643,7 +643,12 @@ def _load_single_patient_merged_variants(self, patient, use_cache=True):
             for patient_variants in patient.variants_list:
                 if type(patient_variants) == str:
                     if ".vcf" in patient_variants:
-                        variant_collections.append(varcode.load_vcf_fast(patient_variants))
+                        try:
+                            variant_collections.append(varcode.load_vcf_fast(patient_variants))
```
Comment (tavinathanson) [inline cohorts/cohort.py:647]: Huh, I wonder why it's showing this when this is already merged?

File: cohorts/io/gcloud_storage.py [hunk 2]
```diff
@@ -119,8 +119,10 @@ def download_to_path(self, gsuri, localpath, binary_mode=False):
                 "No such file on Google Storage: '{}'".format(gs_rel_path))
 
         # A tmp file to serve intermediate phase
-        tmp_fid, tmp_file_path = tempfile.mkstemp(text=(not binary_mode))
-        # Download starts in a sec....
+        tmp_fid, tmp_file_path = tempfile.mkstemp(text=(not binary_mode),
+                                                  dir=tmpdir)
+        # Download starts in a seca....
```
Comment (tavinathanson) [inline cohorts/io/gcloud_storage.py]: Typo: "seca"

File: cohorts/io/gcloud_storage.py [hunk 3]
```diff
@@ -119,8 +119,10 @@ def download_to_path(self, gsuri, localpath, binary_mode=False):
...
+        ablob.chunk_size = 1<<30
```
Comment (tavinathanson) [inline cohorts/io/gcloud_storage.py]: Was confused about this until I saw https://github.com/GoogleCloudPlatform/google-cloud-python/issues/2222, maybe reference it here?

**Comment by tavinathanson** [comment]
LGTM but will defer to @armish until I'm caught up on this nifty new Cohorts addition!

File: cohorts/cohort.py (hunk 1 above)
Comment (jburos) [reply cohorts/cohort.py:647]: I was wondering that too

File: cohorts/io/gcloud_storage.py [hunk 4]
```diff
@@ -64,10 +64,10 @@ def client(self):
         return self._client
 
 
-    def list_files(self, gsuri):
+    def list_files(self, gsuri, **kwargs):
         bucket_name, gs_rel_path = self.parse_uri(gsuri)
         bucket = self._client.get_bucket(bucket_name)
-        return bucket.list_blobs()
+        return bucket.list_blobs(**kwargs)
```
Comment (armish) [inline cohorts/io/gcloud_storage.py:70]: Just curious: when/for what you need to pass additional arguments to the `list_blobs`?  (Never mind - just saw your comment in the other tab). Nice addition here, thanks Jacki!

File: cohorts/io/gcloud_storage.py [hunk 5]
```diff
@@ -119,7 +119,12 @@ def download_to_path(self, gsuri, localpath, binary_mode=False):
                 "No such file on Google Storage: '{}'".format(gs_rel_path))
 
         # A tmp file to serve intermediate phase
-        tmp_fid, tmp_file_path = tempfile.mkstemp(text=(not binary_mode))
+        # should be on same filesystem as localpath
+        tmp_fid, tmp_file_path = tempfile.mkstemp(text=(not binary_mode),
+                                                  dir=tmpdir)
+        # set chunk_size to reasonable default
+        # https://github.com/GoogleCloudPlatform/google-cloud-python/issues/2222
```
Comment (armish) [inline cohorts/io/gcloud_storage.py:126]: Good find!

**Review by armish — APPROVED** [review]
💯 :shipit: 

Thanks a lot for giving this a try and fixing things 🙏

**Pull request merged**